### DISCUSSION TITLE
Printing

### DISCUSSION THREAD

**Discussion by x61** [ahrm/sioyek#618] [General]
Does anyone use sioyek as their default pdf reader? Can you share how you print, files, if needed?  I understand sioyek does not currently have the means of sending files to the printer for printing but can someone share how the navigate around the inability to print directly from sioyek?

**Comment by ahrm** [comment]
Note that you can add something like this to your `prefs_user` (if `chrome.exe` is in your path):

```
new_command _open_in_chrome chrome.exe "%1"
```

Now you can open the document in chrome (or other software that you like) by executing `_open_in_chrome`. You can also bind `_open_in_chrome` to a keybinding in `keys_user.config`.

**Reply by x61** [reply]
This is great. Knowing that I can directly open files in a different pdf viewer is a game changer for me. I am on a mac and unfortunately, the command did not work for me. I added the following command to my `prefs_user.config` file
`new_command _open_in_preview Preview.app "%1"`. and even tried without the extension. I can see the added command, `_open_in_preview` when I typed `:` in sioyek so I do not think I need to "source" or do anything other thank quick and restart sioyek, but it is not opening the file. Any recommendations for macos users?

**Reply by ahrm** [reply]
I don't use mac so I am not sure, but maybe something like this works?

```
new_command _open_in_preview open Preview.app "%1"
```

**Reply by x61** [reply]
Thank you. I tried the command you suggested but still there no success, but I did have the following printed on the terminal:
`Warning: invalid command (key:); resetting to root
The files /Users/mac/Downloads/Preview and /Users/mac/Downloads/"/Users/mac/Downloads/test.pdf" do not exist. `

**Comment by x61** [comment]
In case someone else is interested, here is the solution that worked for me:
`new_command _open_in_preview open -a Preview  %1`.  Thanks @ahrm

**Reply by uncomfyhalomacro** [reply]
For me, it's better to use `lp` command if you are fluent with CUPS

**Reply by x61** [reply]
I liked the idea of opening the doc in a different pdf reader because I sometimes need to sign the file. I am not sure if we can sign documents directly within Sioyek at this time (maybe never).

**Comment by daxmate** [comment]
After using sioyek as the default pdf viewer, the printing function haunts from time to time.

Thanks for the above discussions, here are my worked solutions after several tries(I'm on macOS):
1. **new_command doesn't work**
2. open in Preview
```sh
# in prefs_user
execute_command_v open -a Preview %1  #don't put quotes around %1
# in keys_user
execute_command_v <C-v>
```
3. print file with lp command
```sh
# in prefs_user
execute_command_p /usr/bin/lp -o sides=one-sided -o media=a4 %1
# in keys_user
execute_command_p <C-p>
```

**Comment by duguyipiao** [comment]
If anyone is interested, the utility [gtk-print](https://github.com/nbfalcon/gtk-print) can provide a quite native experience under linux.

For the development version, the following config works for me:
```
# in prefs_user.config
new_command _print gtk-print %1
# in keys_user.config
_print <C-p>
```
For the stable version, use [`execute_command`](https://sioyek-documentation.readthedocs.io/en/latest/configuration.html#execute-command-a-to-execute-command-z) instead.

**Comment by eylles** [comment]
another option for linux is with [YAD](https://github.com/v1cont/yad), yad is available in most distros https://repology.org/project/yad/versions 

this works with the yad version shipped by debian stable
```
# prefs_user.config
execute_command_p yad --print --filename=%1 
# keys_user.config
execute_command_p <C-p>
```

**Reply by boutil** [reply]
Thanks for the tip. I discovered `yad` thanks to your comment. I had to add a `--type=RAW` argument, otherwise it would have tried to print the content of the pdf file, "%PDF-1.7 [...]". Unfortunately the `--type=RAW` argument is incompatible with the otherwise useful `--add-preview`.
